tests/framework/test_time.py

```diff
@@ -56,94 +62,106 @@ def step_column(sim):
     return sim._population._population.step_size
 
 
+def take_step(sim):
+    old_time = sim._clock.time
+    sim.step()
+    new_time = sim._clock.time
+    return new_time - old_time
+
+
 class StepModifier(MockGenericComponent):
-    def __init__(self, name, step_modifier):
+    def __init__(self, name, step_modifier_even, step_modifier_odd):
         super().__init__(name)
-        self.step_modifier = step_modifier
+        self.step_modifier_even = step_modifier_even
+        self.step_modifier_odd = step_modifier_odd
+        self.ts_pipeline_value = None
 
     def setup(self, builder) -> None:
         super().setup(builder)
         builder.value.register_value_modifier("simulant_step_size", self.modify_step)
+        self.rate_pipeline = builder.value.register_value_producer(
+            "test_rate",
+            source=lambda idx: pd.Series(1.75, index=idx),
+            preferred_post_processor=rescale_post_processor,
+        )
 
-    def modify_step(self, index):
-        return pd.Series(pd.Timedelta(days=self.step_modifier), index=index)
-
+    def on_time_step(self, event: Event) -> None:
+        self.ts_pipeline_value = self.rate_pipeline(event.index)
 
-def test_align_times(SimulationContext, base_config, components):
+    def modify_step(self, index):
+        return pd.Series(
+            [
+                pd.Timedelta(days=self.step_modifier_even)
+                if i % 2 == 0
+                else pd.Timedelta(days=self.step_modifier_odd)
+                for i in index
+            ],
+            index=index,
+        )
+
+
+def test_basic_iteration(SimulationContext, base_config, components):
+    base_config["time"]["step_size"] = 1
     sim = SimulationContext(base_config, components)
+    listener = [c for c in components if "listener" in c.args][0]
     sim.setup()
     sim.initialize_simulants()
+    assert sim._clock.time == get_time_stamp(sim.configuration.time.start)
+    assert sim._clock.step_size == pd.Timedelta(days=1)
     pop_size = len(sim.get_population())
     # After initialization, all simulants should be aligned to event times
     assert len(active_simulants(sim)) == pop_size
+    assert np.all(step_pipeline(sim) == step_column(sim))
 
-    sim.step()
-    # After one step (and no step adjustments, simulants should still be aligned)
-    assert len(active_simulants(sim)) == pop_size
-
-    sim._population._population.step_size *= 2
-
-    sim.step()
-    # No simulants should be aligned after a step size adjustment
-    assert active_simulants(sim).empty
-
-    sim.step()
-    # Now they should be aligned again
-    assert len(active_simulants(sim)) == pop_size
+    for _ in range(2):
+        taken_step_size = take_step(sim)
+        # After a step (and no step adjustments, simulants should still be aligned)
+        assert taken_step_size == pd.Timedelta(days=1)
```

Comment:
You might be able to move these assertions into a sort of `validate_aligned` helper method.

Reply:
+1

Reply:
I managed to refactor and clean this up quite a bit, I think